File: odd_power_sets.js
full_set = [1, 2, 3, 4, 5];

all_sets = [];
all_set_combos = Math.pow(2, full_set.length);

for (combo = 0; combo < all_set_combos; combo++) {
    console.log(combo);
    bindigits = '';
    newset = [];
    for (pos = 0; pos < full_set.length; pos++) {
        if (combo & (1 << pos)) {
            bindigits += '1';
            newset.push(full_set[pos]);
        } else {
            bindigits += '0';
        }
    }
    all_sets.push(newset);
    console.log(bindigits);
}

console.log('All sets');
all_sets.forEach(s => {
    console.log(s);
});
console.log('All sets count: ', all_sets.length);

odd_sets = []
all_sets.forEach(s => {
    if (s.length % 2 == 1) {
        odd_sets.push(s);
    }
});

console.log('Odd sets');
odd_sets.forEach(s => {
    console.log(s);
});
console.log('Odd sets count: ', odd_sets.length);

File: OddPowerSets.java
import java.lang.Math;
import java.util.Vector;

class OddPowerSets {
    static int[] full = { 1, 2, 3, 4, 5 };

    public static void main(String[] args) {
        int combo_count = (int) (Math.pow(2, full.length));

        // Build up every set
        Vector<Vector<Integer>> all_sets = new Vector<Vector<Integer>>();
        for (int combo = 0; combo < combo_count; combo++) {
            System.out.println(combo);
            String bincode = "";
            Vector<Integer> newset = new Vector<Integer>();
            for (int c = 0; c < full.length; c++) {
                if ((combo & (1 << c)) > 0) {
                    bincode += "1";
                    newset.addElement(full[c]);
                } else {
                    bincode += "0";
                }
            }
            System.out.println(bincode);
            all_sets.addElement(newset);
        }

        Vector<Vector<Integer>> odd_sets = new Vector<Vector<Integer>>();
        System.out.println("All sets:");
        for (Vector<Integer> s : all_sets) {
            System.out.println(s);
            if (s.size() % 2 == 1) {
                odd_sets.addElement(s);
            }
        }

        System.out.println("Odd sets:");
        for (Vector<Integer> s : odd_sets) {
            System.out.println(s);
        }
        System.out.println("Odd  sets count: " + odd_sets.size());
    }
}
